custom/icds_reports/filters.py

```diff
@@ -66,7 +74,9 @@ def loc_to_json(loc, project):
                 # there are some instances in viewing archived locations where we don't actually
                 # support drilling all the way down.
                 break
-            this_loc['children'] = [loc_to_json(loc, project) for loc in children]
+            this_loc['children'] = [loc_to_json(loc, project) for loc in children if
+                                    user_can_access_location_id(domain, user, loc.location_id) or
```

Comment:
This is going to be a lot of sql queries on each page load

Reply:
Can you help understand what is happening if there is a selected location id, specially `this_loc = [k for k in parent['children'] if k['uuid'] == loc.location_id][0]`?

Reply:
hey yes,
selected ID comes as id of location assigned to the user.
 `this_loc = [k for k in parent['children'] if k['uuid'] == loc.location_id][0]` and the outer loop is actually preparing a dict in tree like structure: 

```

[{
name:<>,
.
.,
children:[
{
name:<>,
children:[...]
}
],
}]

```
the outer loop is applied on the selected_location ancestor hierarchy  only and skips other parent locations

Reply:
I think the thing it is trying to achieve could have been done with recursion. But Not sure why this way was chosen.

Reply:
>  selected ID comes as id of location assigned to the user.

can you confirm that it's not the location selected in the view?